### PR TITLE
forward object not found error in swift as dav 404

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

File: lib/private/Files/ObjectStore/ObjectStoreStorage.php [hunk 1]
```diff
@@ -275,6 +276,12 @@ public function fopen($path, $mode) {
 				if (is_array($stat)) {
 					try {
 						return $this->objectStore->readObject($this->getURN($stat['fileid']));
+					} catch (NotFoundException $e) {
+						$this->logger->logException($e, [
+							'app' => 'objectstore',
+							'message' => 'Count not get object ' . $this->getURN($stat['fileid']) . ' for file ' . $path,
```
Comment (danielkesselberg) [inline lib/private/Files/ObjectStore/ObjectStoreStorage.php]: > Count

:confused:

**Review by MorrisJobke — APPROVED** [review]
Code makes sense 👍

File: lib/private/Files/ObjectStore/ObjectStoreStorage.php (hunk 1 above)
Comment (MorrisJobke) [reply lib/private/Files/ObjectStore/ObjectStoreStorage.php]: Let me fix that one.

**Comment by rullzer** [comment]
Should we also add this for S3?

**Comment by danielkesselberg** [comment]
https://github.com/nextcloud/server/blob/0eebff152a177dd59ed8773df26f1679f8a88e90/lib/public/Files/ObjectStore/IObjectStore.php#L40-L46

For nextcloud15 the interface should be adjusted. `@throws NotFoundException if file does not exist`

**Comment by MorrisJobke** [comment]
> For nextcloud15 the interface should be adjusted. `@throws NotFoundException if file does not exist`

Added 👍

**Review by rullzer — APPROVED** [review]

**Pull request merged**